includes/class-edd-db-customer-meta.php

```diff
@@ -0,0 +1,152 @@
+	 *
+	 * @access  private
+	 * @since   2.6
+	 */
+	public function get_meta( $customer_id = 0, $meta_key = '', $single = false ) {
+		return get_metadata( 'customer', $customer_id, $meta_key, $single );
+	}
+
+	/**
+	 * Add meta data field to a customer.
+	 *
+	 * For internal use only. Use EDD_Customer->add_meta() for public usage.
+	 *
+	 * @param   int    $customer_id   Customer ID.
+	 * @param   string $meta_key      Metadata name.
+	 * @param   mixed  $meta_value    Metadata value.
+	 * @param   bool   $unique        Optional, default is false. Whether the same key should not be added.
+	 * @return  bool                  False for failure. True for success.
+	 *
+	 * @access  private
+	 * @since   2.6
+	 */
+	public function add_meta( $customer_id = 0, $meta_key = '', $meta_value, $unique = false ) {
+		return add_metadata( 'customer', $customer_id, $meta_key, $meta_value, $unique );
+	}
+
+	/**
+	 * Update customer meta field based on Customer ID.
+	 *
+	 * For internal use only. Use EDD_Customer->update_meta() for public usage.
+	 *
+	 * Use the $prev_value parameter to differentiate between meta fields with the
+	 * same key and Customer ID.
+	 *
+	 * If the meta field for the customer does not exist, it will be added.
+	 *
+	 * @param   int    $customer_id   Customer ID.
+	 * @param   string $meta_key      Metadata key.
+	 * @param   mixed  $meta_value    Metadata value.
+	 * @param   mixed  $prev_value    Optional. Previous value to check before removing.
+	 * @return  bool                  False on failure, true if success.
+	 *
+	 * @access  private
+	 * @since   2.6
+	 */
+	public function update_meta( $customer_id = 0, $meta_key = '', $meta_value, $prev_value = '' ) {
+		return update_metadata( 'customer', $customer_id, $meta_key, $meta_value, $prev_value );
+	}
+
+	/**
+	 * Remove metadata matching criteria from a customer.
+	 *
+	 * For internal use only. Use EDD_Customer->delete_meta() for public usage.
+	 *
+	 * You can match based on the key, or key and value. Removing based on key and
+	 * value, will keep from removing duplicate metadata with the same key. It also
+	 * allows removing all metadata matching key, if needed.
+	 *
+	 * @param   int    $customer_id   Customer ID.
+	 * @param   string $meta_key      Metadata name.
+	 * @param   mixed  $meta_value    Optional. Metadata value.
+	 * @return  bool                  False for failure. True for success.
+	 *
+	 * @access  private
+	 * @since   2.6
+	 */
+	public function delete_meta( $customer_id = 0, $meta_key = '', $meta_value = '' ) {
+		return delete_metadata( 'customer', $customer_id, $meta_key, $meta_value );
+	}
+
+	/**
+	 * Create the table
+	 *
+	 * @access  public
+	 * @since   2.6
+	*/
+	public function create_table() {
+
+		require_once( ABSPATH . 'wp-admin/includes/upgrade.php' );
+
+		$sql = "CREATE TABLE {$this->table_name} (
+			meta_id bigint(20) NOT NULL AUTO_INCREMENT,
+			customer_id bigint(20) NOT NULL DEFAULT '0',
```

Comment:
So after a little more reading, while an accepted item in the `create` it's not applied, just parsed. We will just have to handle this within the classes, which I'll push up here shortly.